improver/cube_combiner.py

```diff
@@ -140,6 +167,57 @@ def _get_expanded_coord_names(cube_list):
                     expanded_coords.append(coord)
         return expanded_coords
 
+    def _setup_coord_broadcast(self, cube_list):
+        """
+        Adds a scalar DimCoord to any subsequent cube in cube_list so that they all include all of
+        the coords specified in self.broadcast_coords in the right order.
+
+        Args:
+            cube_list: (iris.cube.CubeList)
+
+        Returns:
+            iris.cube.CubeList
+                Updated version of cube_list
+
+        """
+        for coord in self.broadcast_coords:
+            target_cube = cube_list[0]
+            try:
+                if coord == "threshold":
+                    target_coord = find_threshold_coordinate(target_cube)
+                else:
+                    target_coord = target_cube.coord(coord)
+            except CoordinateNotFoundError:
+                raise CoordinateNotFoundError(f"{coord} not found in {target_cube}")
+            new_list = CubeList([])
+            for cube in cube_list:
+                try:
+                    found_coord = cube.coord(target_coord)
+                except CoordinateNotFoundError:
+                    new_coord = DimCoord(
+                        [0],
+                        standard_name=target_coord.standard_name,
+                        long_name=target_coord.long_name,
+                        var_name=target_coord.var_name,
+                        units=target_coord.units,
+                        bounds=None,
+                        attributes=target_coord.attributes,
+                        coord_system=target_coord.coord_system,
+                    )
+                    cube.add_aux_coord(new_coord, None)
+                    cube = iris.util.new_axis(cube, new_coord)
+                    enforce_coordinate_ordering(
+                        cube, [d.name() for d in target_cube.coords(dim_coords=True)]
+                    )
+                else:
+                    if found_coord not in cube.dim_coords:
+                        raise TypeError(
+                            f"Cannot broadcast to coord {coord} as it already exists as an AuxCoord"
+                        )
```

Comment:
In this case, why is it not appropriate to promote the existing AuxCoord to a dimension?

Reply:
If the coordinate already exists, then it means that the cube only applies to a subset of that coordinate, not the whole coordinate, in which case it probably isn't appropriate to use the cube for that purpose.
I did consider checking whether the bounds on that AuxCoord completely span the coordinate it is being broadcast to, but I couldn't think of a use-case, so left it like this.

Reply:
OK, that makes sense.  I think maybe a short comment would help the user here, but feel free to disagree.

Reply:
Comment added